packages/cli/workspace-loader/src/workspaces/FernWorkspace.ts

```diff
@@ -48,3 +50,113 @@ export class FernWorkspace extends AbstractAPIWorkspace<void> {
         return this;
     }
 }
+
+export declare namespace LazyFernWorkspace {
+    export interface BaseArgs {
+        workspaceName: string | undefined;
+        absoluteFilepath: AbsoluteFilePath;
+        generatorsConfiguration: generatorsYml.GeneratorsConfiguration | undefined;
+        changelog: APIChangelog | undefined;
+    }
+
+    export interface Args extends BaseArgs {
+        context: TaskContext;
+        cliVersion: string;
+    }
+}
+
+export class LazyFernWorkspace extends AbstractAPIWorkspace<OSSWorkspace.Settings> {
+    public type: "fern" | "oss" = "fern";
+    public workspaceName: string | undefined;
+    public absoluteFilepath: AbsoluteFilePath;
+    public generatorsConfiguration: generatorsYml.GeneratorsConfiguration | undefined;
+    public changelog: APIChangelog | undefined;
+
+    private context: TaskContext;
+    private cliVersion: string;
+
+    constructor({
+        absoluteFilepath,
+        workspaceName,
+        generatorsConfiguration,
+        changelog,
+        cliVersion,
+        context
+    }: LazyFernWorkspace.Args) {
+        super();
+        this.absoluteFilepath = absoluteFilepath;
+        this.workspaceName = workspaceName;
+        this.changelog = changelog;
+        this.generatorsConfiguration = generatorsConfiguration;
+        this.cliVersion = cliVersion;
+        this.context = context;
+    }
+
+    public async getDefinition(
+        { context }: { context?: TaskContext },
+        settings?: OSSWorkspace.Settings
+    ): Promise<FernDefinition> {
+        return (await this.toFernWorkspace({ context }, settings)).definition;
+    }
+
+    public async toFernWorkspace(
```

Comment:
@armandobelardo can we store a private member on this to see if this function was already called? if already called, would be great to not log anything

Reply:
Yeah def!